kinto/core/storage/postgresql/schema.sql

```diff
@@ -95,6 +95,34 @@ END;
 $$ LANGUAGE plpgsql;
 
 --
+-- Find timestamp in case of conflict
+--
+
+CREATE OR REPLACE FUNCTION find_timestamp(pid VARCHAR, cid VARCHAR, ts TIMESTAMP)
+RETURNS TIMESTAMP AS $$
+DECLARE
+    previous_collection_ts TIMESTAMP;
+    record_ts TIMESTAMP;
+BEGIN
+	record_ts := ts;
+
+	previous_collection_ts := NULL;
+    SELECT last_modified INTO previous_collection_ts
+      FROM timestamps
+     WHERE parent_id = pid
+       AND collection_id = cid;
+
+    IF ts IS NULL OR
+       (previous_collection_ts IS NOT NULL AND as_epoch(ts) = as_epoch(previous_collection_ts)) THEN
```

Comment:
If I'm not wrong, `previous_collection_ts` will never be NULL because it would have been set during the first operation before the conflict occured.


Also, if I understand correctly, this will re-run the trigger with `record.last_modified = collection_ts`, which will re-bump the timestamp (as if a user would have had specified it in a normal PUT). Hence, shouldn't we compare with `as_epoch(ts) <= as_epoch(previous_collection_ts))`